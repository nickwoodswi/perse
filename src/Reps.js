import React, { Component } from 'react'

class Reps extends Component {
    render() {
        if (this.props.reptype === 'TO FAILURE') {
            return(<> (TO FAILURE)<br/></>)
        }
        if (this.props.reptype === 'SET MULTIPLE') {
            return(<> <b>x {this.props.reps}</b><br/></>)
        }
        if (this.props.reptype === 'SINGLE DISTANCE (m)') {
            return(<> <b>{this.props.reps}m</b><br/></>)
        }
        if (this.props.reptype === 'TIME (min)') {
            return(<> <b>{this.props.reps}min</b></>)
        }
    }
}

export default Reps