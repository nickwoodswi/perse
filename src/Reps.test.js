import React from 'react';
import { render } from '@testing-library/react';
import Reps from './Reps';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders Reps component', () => {
    render(<MemoryRouter><Reps /></MemoryRouter>);
});

it('Reps component renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><Reps /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});