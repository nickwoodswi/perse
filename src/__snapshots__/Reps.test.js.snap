// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reps component renders the UI as expected 1`] = `" No rep type selected"`;
